### DISCUSSION TITLE
Why can't I set gaps for specific windows. "for_window [class="vlc"] gaps ANYVALUE 0" config not applying

### DISCUSSION THREAD

**Discussion by ImGGAAVVIINN** [i3/i3#6263] [Q&A] [answered]
In my config, I have
for_window [class="vlc"] gaps top 0
for_window [class="Nemo"] gaps bottom 0

but my gaps is still looking like this:
![Screenshot_20241012_175704](https://github.com/user-attachments/assets/654662bb-8d0c-48ea-8692-44e0297fc965)
Am I doing something wrong?

**Comment by okraits** [comment]
Gaps can only be set per workspace and not per window. See https://i3wm.org/docs/userguide.html#changing_gaps

**Comment by ImGGAAVVIINN** [comment]
But it has "for_window [class="vlc"] gaps inner 0, gaps outer 0" under the
examples and I copied it

**Reply by okraits** [reply]
Hmm, interesting. AFAIK the syntax of the `gaps` command in that `for_window` example isn't even correct. I would have to look into the code to check how the behaviour is in reality. I guess @orestisfl or @stapelberg could answer that easily.

**Reply by stapelberg** [reply, marked as answer]
Hmm, it should probably be `gaps inner current set 0` and `gaps outer current set 0`, I suppose? Can you give that a try and send a PR to update the example if so? Thanks.

**Reply by ImGGAAVVIINN** [reply]
Is it possible to achieve my desired result of turning off gaps for specific windows with ```gaps inner current set 0```?
Edit: yes it works. but I am unable to apply these settings to just top or bottom of a window, is it currently possible to do so?
